clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/factory/CachingDaVinciClientFactory.java

```diff
@@ -123,7 +123,9 @@ public synchronized void close() {
     LOGGER.info("Closing client factory, clientCount={}", clients.size());
     for (DaVinciClient client: clients) {
       try {
-        client.close();
+        if (client != null) {
```

Comment:
Will remove, leftover code from experimentation while debugging.

Reply:
Although if the user closed the client already this will `throw new VeniceClientException("Da Vinci client is not ready, storeName=" + getStoreName());` Do we want to check and prevent exception?

Reply:
But we a catching and logging it already, I think we're fine here.